extensions/schema-compare/src/schemaCompareMainWindow.ts

```diff
@@ -386,16 +387,63 @@ export class SchemaCompareMainWindow {
 		this.tablelistenersToDispose.push(this.differencesTable.onCellAction(async (rowState) => {
 			let checkboxState = <azdata.ICheckboxCellActionEventArgs>rowState;
 			if (checkboxState) {
+				this.differencesTable.checked = undefined;
+				this.differencesTable.selectedRows = [checkboxState.row];
 				let diff = this.comparisonResult.differences[checkboxState.row];
-				await service.schemaCompareIncludeExcludeNode(this.comparisonResult.operationId, diff, checkboxState.checked, azdata.TaskExecutionMode.execute);
-				this.saveExcludeState(checkboxState);
+				const result = await service.schemaCompareIncludeExcludeNode(this.comparisonResult.operationId, diff, checkboxState.checked, azdata.TaskExecutionMode.execute);
+				if (result.success) {
+					this.saveExcludeState(checkboxState);
+
+					// depencies could have been included or excluded as a result, so save their exclude states
+					result.affectedDependencies.forEach(difference => {
+						// find the row of the difference and set it's checkbox
+						const row = this.findDifferenceRow(difference);
+						if (row !== -1) {
+							this.differencesTable.checked = { row: row, columnName: 'Include', checked: difference.included };
+							const dependencyCheckBoxState: azdata.ICheckboxCellActionEventArgs = {
+								checked: difference.included,
+								row: row,
+								column: 2,
+								columnName: undefined
+							};
+							this.saveExcludeState(dependencyCheckBoxState);
+						}
+					});
+				} else {
+					// failed because of dependencies
+					if (result.blockingDependencies) {
+						// show the first dependent that caused this to fail in the warning message
+						const firstDependentName = this.createName(result.blockingDependencies[0].sourceValue ? result.blockingDependencies[0].sourceValue : result.blockingDependencies[0].targetValue);
+						vscode.window.showWarningMessage(checkboxState.checked ? cannotIncludeMessage + firstDependentName : cannotExcludeMessage + firstDependentName);
```

Comment:
Can we use paratermets in loc string to be consistent... also am not sure how "+" would work in all languages if the ordering insentence changes...
localize('schemaCompare.cannotExcludeMessage', "Cannot exclude. Included dependents exist such as {0}", firstDependentName);

Reply:
Fixed :)